doc/factors.md

```diff
@@ -0,0 +1,208 @@
+# Using marian with factors
+
+Following this README should allow the user to train a model with source- and/or target-side factors. To train with factors, the data must be formatted in a certain way. A special vocabulary file format is also required. See details below.
+
+### Requirements:
+
+Marian from this repository.
+
+## Define factors
+
+Factors should be organized in "groups," where each group represents a different feature. For example, there could be a group denoting capitalization and another denoting subword divisions.
+
+Factors within a single group should start with the same string.
+
+For example, for a capitalization factor group, the individual factors could be:
+
+`c0`: all lowercase
+
+`c1`: first character capitalized, rest lowercase
+
+`c2`: all uppercase
+
+If there were a second factor group for subword divisions, the individual factors could be:
+
+`s0`: end of word, whitespace should follow
+
+`s1`: join token with next subword
+
+There is no limit on the number of factor groups barring some practical limitations having to do with how the vocabulary is stored by `marian`. If the limit is exceeded `marian` will throw this error.
+
+Factor group zero is always the actual words in the text, referred to as *lemmas*.
+
+## Data preparation
+
+Factors are appended to the *lemmas* with a pipe `|`. The pipe also separates factors of multiple groups.
+
+Example sentence:
+
+```
+Trump tested positive for COVID-19.
+```
+
+Preprocessed sentence:
+```
+trump test@@ ed positive for c@@ o@@ v@@ i@@ d - 19 .
+```
+
+Apply factors:
+```
+trump|c1|s0 test|c0|s1 ed|c0|s0 positive|c0|s0 for|c0|s0 c|c2|s1 o|c2|s1 v|c2|s1 i|c2|s1 d|c2|s0 -|c0|s0 19|c0|s0 .|c0|s0
+```
+
+
+## Create the factored vocabulary
+
+Factored vocabularies should have the extension `.fsv`. How to structure the vocabulary file is described below. If using factors only on the source or target side, the vocabulary of the other side can be a normal `json`, `yaml`, etc.
+
+The `.fsv` vocabulary should have three sections:
```

Comment:
The third section that is missing was the 'factor distribution section'. This was removed from the documentation because its not needed for now based on the current state of the code https://github.com/marian-nmt/marian-dev/blob/c228c77a527a30488e2279a90111c3f8906bc7eb/src/data/factored_vocab.cpp#L68-L72 
As you can see only checks are made and nothing is stored anywhere. Changed `three sections` to `two sections`.

Reply:
Thanks. Please change also "should have the extension" to "must have the extension" as it is strictly required.